admin-dev/themes/new-theme/js/pages/order/view/order-view-page.js

```diff
@@ -337,4 +285,62 @@ export default class OrderViewPage {
   getActivePage() {
     return $(OrderViewPageMap.productsTablePagination).find('.active span').get(0);
   }
+
+  refreshProductsList(orderId) {
+    const $loadingSpinner = $(OrderViewPageMap.productsPanel).find('.spinner-order-products-container#orderProductsLoading');
+    $loadingSpinner.show();
+
+    const $tablePagination = $(OrderViewPageMap.productsTablePagination);
+    const numRowsPerPage = $tablePagination.data('numPerPage');
+    const initialNumProducts = $(OrderViewPageMap.productsTableRows).length;
+    let currentPage = parseInt($(OrderViewPageMap.productsTablePaginationActive).html(), 10);
+
+    $.ajax(this.router.generate('admin_orders_get_products', {orderId}))
+        .done((response) => {
+          // Delete previous product lines
+          $(OrderViewPageMap.productsTable).find(OrderViewPageMap.productsTableRows).remove();
+
+          $(OrderViewPageMap.productsTable + ' tbody').prepend(response);
+
+          $loadingSpinner.hide();
+
+          const newNumProducts = $(OrderViewPageMap.productsTableRows).length;
+          const newPagesNum = Math.ceil(newNumProducts / numRowsPerPage);
+
+          this.orderProductRenderer.updateNumProducts(newNumProducts);
+          this.orderProductRenderer.initPagination();
+
+          // Display alert
+          if (initialNumProducts > newNumProducts) { // product deleted
+            $.growl.notice({
+              title: '',
+              message: window.translate_javascripts['Items successfully removed']
+                  .replace('[1]', (initialNumProducts-newNumProducts))
+            });
+            // Move to page of the deleted item
+            EventEmitter.emit(OrderViewEventMap.productListPaginated, {
```

Comment:
This should be performed outside of this `if/elseif` block because in case `initialNumProducts == newNumProducts` (for an unknown weird reason) the event is never triggered

So you can init `let numPage = 1` before the id, only change its value when `initialNumProducts > newNumProducts` and then always trigger the event after the growl So that we are sure to be in a consistent state